waspstation/code/modules/mining/deepcore/ore_vein.dm

```diff
@@ -0,0 +1,37 @@
+GLOBAL_LIST_EMPTY(ore_vein_landmarks)
+
+/obj/effect/landmark/ore_vein
+	name = "ore vein"
+	var/datum/material/resource
+	var/material_rate = 0
+
+/obj/effect/landmark/ore_vein/Initialize(mapload, var/datum/material/mat)
+	. = ..()
+	GLOB.ore_vein_landmarks += src
+	// Key = Material path; Value = Material Rate
+	//! Ensure material datum has an ore_type set
+	var/static/list/ores_list = list(
```

Comment:
This setup + the way these are spawned looks to me like it doesn't necessarily guarantee that one of each type spawns. Is that accurate?

Reply:
Correct, however the number of veins spawn near ~100 so we can probably just assume a even distribution.